PostHog/Autocapture/AutocaptureEventProcessing.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  AutocaptureEventProcessing.swift
+//  PostHog
+//
+//  Created by Yiannis Josephides on 22/10/2024.
+//
+
+#if os(iOS) || targetEnvironment(macCatalyst)
+    import UIKit
+
+    protocol AutocaptureEventProcessing: AnyObject {
+        func process(source: PostHogAutocaptureIntegration.EventData.EventSource, event: PostHogAutocaptureIntegration.EventData)
+    }
+
+    class PostHogAutocaptureEventProcessor: AutocaptureEventProcessing {
+        private weak var postHogInstance: PostHogSDK?
+        private weak var postHogConfig: PostHogConfig?
+        private var debounceTimers: [Int: Timer] = [:]
+
+        init(postHogInstance: PostHogSDK, configuration: PostHogConfig) {
+            self.postHogInstance = postHogInstance
+            self.postHogConfig = configuration
+            PostHogAutocaptureIntegration.addEventProcessor(self)
+        }
+
+        deinit {
+            PostHogAutocaptureIntegration.removeEventProcessor(self)
+        }
```

Comment:
I'd like to understand a bit better how those event processors work, we can discuss this after the daily over a call tomorrow?